pkg/printers/internalversion/printers.go

```diff
@@ -419,19 +422,57 @@ func AddHandlers(h printers.PrintHandler) {
 	h.TableHandler(controllerRevisionColumnDefinition, printControllerRevision)
 	h.TableHandler(controllerRevisionColumnDefinition, printControllerRevisionList)
 
+	// Addding known resources beforehand - cf: pkg/printers/humanreadable.go.
+	h.TableHandler(objectMetaColumnDefinitions, printAPIService)
+	h.TableHandler(objectMetaColumnDefinitions, printPodPreset)
+	h.TableHandler(objectMetaColumnDefinitions, printCustomResourceDefinition)
+	h.TableHandler(objectMetaColumnDefinitions, printResourceQuota)
+	h.TableHandler(objectMetaColumnDefinitions, printLimitRange)
+	h.TableHandler(objectMetaColumnDefinitions, printClusterRole)
+	h.TableHandler(objectMetaColumnDefinitions, printRole)
+
 	AddDefaultHandlers(h)
 }
 
+// types without defined columns
+var objectMetaColumnDefinitions = []metav1beta1.TableColumnDefinition{
+	{Name: "Name", Type: "string", Format: "name", Description: metav1.ObjectMeta{}.SwaggerDoc()["name"]},
+	{Name: "Age", Type: "string", Description: metav1.ObjectMeta{}.SwaggerDoc()["creationTimestamp"]},
+}
+
 // AddDefaultHandlers adds handlers that can work with most Kubernetes objects.
 func AddDefaultHandlers(h printers.PrintHandler) {
-	// types without defined columns
-	objectMetaColumnDefinitions := []metav1beta1.TableColumnDefinition{
-		{Name: "Name", Type: "string", Format: "name", Description: metav1.ObjectMeta{}.SwaggerDoc()["name"]},
-		{Name: "Age", Type: "string", Description: metav1.ObjectMeta{}.SwaggerDoc()["creationTimestamp"]},
-	}
 	h.DefaultTableHandler(objectMetaColumnDefinitions, printObjectMeta)
 }
 
+func printAPIService(obj *apiregistration.APIService, options printers.PrintOptions) ([]metav1beta1.TableRow, error) {
+	return printObjectMeta(obj, options)
```

Comment:
registering default handlers like this just so the types show up in compiled-in completions is not the direction the CLI is going. we are more likely to switch to using discovered resource names to drive completion and remove reliance on internal printers entirely